Tasks/DotNetCoreInstallerV1/versioninstaller.ts

```diff
@@ -22,64 +22,60 @@ export class VersionInstaller {
     }
 
     public async downloadAndInstall(versionInfo: VersionInfo, downloadUrl: string): Promise<void> {
-        if (!versionInfo || !versionInfo.version || !url.parse(downloadUrl)) {
-            throw tl.loc("VersionCanNotBeDownloadedFromUrl", versionInfo, downloadUrl);
-        }
-
-        let downloadPath = "";
-        let version = versionInfo.version;
-
-        try {
-            downloadPath = await toolLib.downloadTool(downloadUrl);
-        } catch (error) {
-            throw tl.loc("CouldNotDownload", downloadUrl, JSON.stringify(error));
-        }
-
-        try {
-            //todo: if installation path is outside agents directory, acquire Lock for installation and start timer of 10-20 minutes, after which lock shall be auto released.
-
-            //todo when lock work is done: Check if already installed
-            // this.isVersionInstalled(version);
-
-            // Extract
-            console.log(tl.loc("ExtractingPackage", downloadPath));
-            let extPath: string = tl.osType().match(/^Win/) ? await toolLib.extractZip(downloadPath) : await toolLib.extractTar(downloadPath);
-
-            // Copy folders
-            tl.debug(tl.loc("CopyingFoldersIntoPath", this.installationPath));
-            var allRootLevelEnteriesInDir: string[] = tl.ls("", [extPath]).map(name => path.join(extPath, name));
-            var directoriesTobeCopied: string[] = allRootLevelEnteriesInDir.filter(path => fs.lstatSync(path).isDirectory());
-            directoriesTobeCopied.forEach((directoryPath) => {
-                tl.cp(directoryPath, this.installationPath, "-rf", false);
-            });
-
-            // Copy files
-            try {
-                if (this.packageType == utils.Constants.sdk && this.isLatestInstalledVersion(version)) {
-                    tl.debug(tl.loc("CopyingFilesIntoPath", this.installationPath));
-                    var filesToBeCopied = allRootLevelEnteriesInDir.filter(path => !fs.lstatSync(path).isDirectory());
-                    filesToBeCopied.forEach((filePath) => {
-                        tl.cp(filePath, this.installationPath, "-f", false);
-                    });
-                }
-            }
-            catch (ex) {
-                tl.warning(tl.loc("FailedToCopyTopLevelFiles", this.installationPath, JSON.stringify(ex)));
+        return new Promise<void>((resolve, reject) => {
```

Comment:
any reason for changing the pattern from await to using promise chain

Reply:
It required use of a lot of individual try catch, was looking clumsy to me, hence changed to promise chained code.

Reply:
reverted to use await rather